tensorboard/program.py

```diff
@@ -597,6 +597,7 @@ def __init__(self, wsgi_app, flags):
             host = "localhost"
 
         self._host = host
+        self.display_host = None  # Will be set by get_url() below
```

Comment:
Let's just cache the entire URL for this, so `self._url`.

That way we can skip even the last formatting step in `get_url()` if we already have generated the URL.

Reply:
Yes, I cached the entire URL in 3f99276.